pkg/controller/cheinstallation/cheinstallation_controller_test.go

```diff
@@ -475,6 +481,49 @@ func TestReconcile(t *testing.T) {
 			AssertThatCheCluster(t, cheCluster.Namespace, cheCluster.Name, cl).DoesNotExist()
 			AssertThatCheInstallation(t, cheInstallation.Namespace, cheInstallation.Name, cl).HasConditions(InstallationFailed("checlusters.org.eclipse.che \"codeready-workspaces\" not found"))
 		})
+
+		t.Run("should delete CheCluster when deleting CheInstallation", func(t *testing.T) {
+			// given
+			cheInstallation := NewInstallation()
+			deletionTS := metav1.NewTime(time.Now())
+			cheInstallation.SetDeletionTimestamp(&deletionTS) // mark resource as deleted
+			cheOperatorNS := cheInstallation.Spec.CheOperatorSpec.Namespace
+			cheCluster := NewCheCluster(cheOperatorNS)
+			cl, r := configureClient(t, cheInstallation,
+				newCheNamespace(cheOperatorNS, v1.NamespaceActive),
+				NewOperatorGroup(cheOperatorNS),
+				NewSubscription(cheOperatorNS),
+				cheCluster)
+			// make sure that the client sets the `deletionTimeStamp` when a CheCluster resource is deleted
+			cl.MockDelete = func(ctx context.Context, obj runtime.Object, opts ...client.DeleteOption) error {
+				fmt.Printf("deleting resource of type '%T'\n", obj)
+				if obj, ok := obj.(*orgv1.CheCluster); ok {
+					// mark namespaces as deleted...
+					deletionTS := metav1.NewTime(time.Now())
+					obj.SetDeletionTimestamp(&deletionTS)
+					// ... but replace them in the fake client cache yet instead of deleting them
+					return cl.Client.Update(ctx, obj)
+				}
+				return cl.Client.Delete(ctx, obj, opts...)
+			}
```

Comment:
do we need to really care about the timestamp - couldn't we just let the client delete the resource? To verify that in the `// then` part, we would just check if the resource is gone. And we should also check that the finalizer is gone - in the similar way as it is done for example done for MUR GC: https://github.com/codeready-toolchain/host-operator/blob/master/pkg/controller/masteruserrecord/masteruserrecord_controller_test.go#L497-L519

Reply:
yeah, right. My intent was to make sure that the CheCluster resource was "marked for deletion" (assuming that in a real case, it would be processed by the Che operator afterwards). But we could also just check that it's deleted and not available in the client cache at the end of the test, indeed. 

And yes, I will also verify that the CheInstallation resource has no finalizer. Good point!

Reply:
done in 6030df0